substrate/frame/timestamp/src/lib.rs

```diff
@@ -368,6 +368,7 @@ impl<T: Config> UnixTime for Pallet<T> {
 		// `sp_timestamp::InherentDataProvider`.
 		let now = Now::<T>::get();
 
+		#[cfg(not(feature = "std"))]
```

Comment:
```
			if now == T::Moment::zero() {
             log::error!(
			target: "runtime::timestamp",
			"`pallet_timestamp::UnixTime::now` is called at genesis, invalid value returned: 0",
		);
}
```

This is the proper fix. The linked pr omitted the if..

Reply:
Added the if statement back, thanks.